core/dbt/adapters/base/column.py

```diff
@@ -108,3 +110,46 @@ def numeric_type(cls, dtype: str, precision: Any, scale: Any) -> str:
 
     def __repr__(self) -> str:
         return "<Column {} ({})>".format(self.name, self.data_type)
+
+    @classmethod
```

Comment:
While we're in here, can we override the `string_size` on the `SnowflakeColumn` class? I think right now, the base logic looks like:
```
    def string_size(self) -> int:
        if not self.is_string():
            raise RuntimeException("Called string_size() on non-string field!")

        if self.dtype == 'text' or self.char_size is None:
            # char_size should never be None. Handle it reasonably just in case
            return 256
        else:
            return int(self.char_size)
```

But on Snowflake, we want to map `TEXT`, `STRING`, or `VARCHAR` (with no size specified) to `VARCHAR(MAX)`, which is [16777216](https://docs.snowflake.com/en/sql-reference/data-types-text.html#varchar).